src/components/fields/date-field/date-field.tsx

```diff
@@ -111,20 +112,35 @@ export const DateField = (props: IGenericFieldProps<IDateFieldSchema>) => {
 						if (!isValidDate(value) || !excludedDatesRule) return true;
 						return !excludedDatesRule["excludedDates"].includes(value);
 					}
+				)
+				.test(
+					"within-days",
+					withinDaysRule?.errorMessage ||
+						(withinDaysRule?.["withinDays"] &&
+							ERROR_MESSAGES.DATE.WITHIN_DAYS(withinDaysRule?.["withinDays"])),
+					(value) => {
+						if (!isValidDate(value) || !withinDaysRule) return true;
+						return DateTimeHelper.checkWithinDays(value, { ...withinDaysRule["withinDays"], dateFormat });
+					}
 				),
 			validation
 		);
 
 		// set minDate / maxDate / disabledDates props
+		const withinDaysRange =
+			withinDaysRule?.["withinDays"] &&
+			DateTimeHelper.calculateWithinDaysRange({ ...withinDaysRule["withinDays"], dateFormat });
 		const minDateProp = getLatestDate([
 			minDate,
 			futureRule?.["future"] && LocalDate.now().plusDays(1),
 			notPastRule?.["notPast"] && LocalDate.now(),
+			withinDaysRule?.["withinDays"] && withinDaysRange.startDate,
```

Comment:
`withinDaysRange` already got the truthy check for `withinDaysRule?.["withinDays"]`

```suggestion
			withinDaysRange?.startDate,
```